api/v1alpha1/hostedcluster_types.go

```diff
@@ -97,8 +97,6 @@ type HostedClusterSpec struct {
 	// will be used to associate various cloud resources with the HostedCluster
 	// and its associated NodePools.
 	//
-	// TODO(dan): consider moving this to .platform.aws.infraID
-	//
 	// +immutable
 	InfraID string `json:"infraID"`
```

Comment:
should be optional now, no? (and not marked as +immutable)

Reply:
Well it is immutable after it has been set once

Reply:
sure, I guess it's a grey area :) ... it can potentially mutate at most once after creation